libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/ExecutionOptions.kt

```diff
@@ -81,4 +90,15 @@ interface MutableExecutionOptions<T> : ExecutionOptions {
   fun enableAutoPersistedQueries(enableAutoPersistedQueries: Boolean?): T
 
   fun canBeBatched(canBeBatched: Boolean?): T
+
+  /**
+   * Configures whether exceptions such as cache miss or other [ApolloException] should throw, instead of being emitted in
+   * [ApolloResponse.exception].
+   *
+   * If true, the call site must catch [ApolloException]. This was the behavior in Apollo Kotlin 3.
+   *
+   * Default: false
+   */
+  @Deprecated("Provided as a convenience to migrate from 3.x, will be removed in a future version")
```

Comment:
Actually now I'm wondering if it should be deprecated as it could be handy in certain cases. For instance with subscriptions, when receiving an error message, in the doc we show [how to use retryWhen](https://www.apollographql.com/docs/kotlin/advanced/websocket-errors#websocket-errors-with-single-consumers), which no longer works, but would with `throwOnException(true)`.

Reply:
OTOH it's easy for a project to implement it themselves, so let's not clutter the API surface